spring-cloud-gcp-data-datastore/src/test/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplateTests.java

```diff
@@ -538,10 +537,20 @@ public void saveTest() {
 	}
 
 	@Test
-	public void saveTestLazy() {
+	public void saveTestCollectionLazy() {
 		this.ob1.lazyMultipleReference = LazyUtil.wrapSimpleLazyProxy(
-				(keys) -> Collections.singletonList(this.childEntity7), List.class,
-				Collections.singletonList(KeyValue.of(this.childKey7)));
+				() -> Collections.singletonList(this.childEntity7), List.class,
+				ListValue.of(KeyValue.of(this.childKey7)));
+		saveTestCommon(this.ob1, true);
+	}
+
+	@Test
+	public void saveTestNotInterfaceLazy() {
+		ArrayList<ChildEntity> arrayList = new ArrayList();
```

Comment:
Is this going to also go through the collection path (similar to saveTestCollectionLazy)?

Reply:
no, ArrayList is not an interface, so it will be proxied using cglib